pkg/codegen/nodejs/gen.go

```diff
@@ -2788,6 +2815,51 @@ func (mod *modContext) genUtilitiesFile(w io.Writer) error {
 	}
 
 	_, err = fmt.Fprintf(w, "%s", code)
+	if err != nil {
+		return err
+	}
+
+	if def.Parameterization != nil {
+
+		parameterValue := fmt.Sprintf("Uint8Array.from(atob(%q), c => c.charCodeAt(0))", base64.StdEncoding.EncodeToString(def.Parameterization.Parameter))
+
+		_, err = fmt.Fprintf(w, `
+const _packageLock = new mutex.Mutex();
+var _packageRef : undefined | string = undefined;
+export async function getPackage() : Promise<string | undefined> {
+	if (_packageRef === undefined) {
+		await _packageLock.acquire();
+		if (_packageRef === undefined) {
+			const monitor = runtime.getMonitor();
+			const params = new resproto.Parameterization();
+			params.setName("%s");
+			params.setVersion("%s");
+			params.setValue(%s);
+
+			const req = new resproto.RegisterPackageRequest();
+			req.setName("%s");
+			req.setVersion("%s");
+			req.setDownloadUrl("%s");
+			req.setParameterization(params);
+			const resp : any = await new Promise((resolve, reject) => {
+				monitor!.registerPackage(req, (err: any, resp: any) => {
```

Comment:
I'm not familiar with the engine-side of this. Is this idempotent?

Reply:
It is